magento/magento2
# Remove 'pub/' directory from paths in css source maps when compiling using grunt

### Description (*)
When compiling css using grunt, and generating sourcemaps, the url's in the sourcemap point to .less files inside the `pub/` directory, which is not correct, as Magento's document root is set to `pub/` in the webserver config, so `pub/` should not be in the url of the sourcemaps.

Fixed with the help from reading the [documentation from less](https://lesscss.org/usage/#less-options-source-map-basepath):
> #### Source Map Basepath
>
> `lessc --source-map-basepath=less-files/`	`{ sourceMap: { sourceMapBasepath: 'less-files/' } }`
>
> This is the opposite of the rootpath option, it specifies a path which should be removed from the output paths. For instance if you are compiling a file in the less-files directory but the source files will be available on your web server in the root or current directory, you can specify this to remove the additional `less-files` part of the path.
> 
> It defaults to the path to the input less file.

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/magento2/issues/38837

### Manual testing scenarios (*)
1. Setup clean Magento
2. Run:
```
$ cp package.json.sample package.json
$ cp Gruntfile.js.sample Gruntfile.js
$ npm install
$ grunt clean:luma && grunt exec:luma && grunt less:luma
```
3. Open the frontend in your browser and open the inspector
4. Expectation is not to see a `pub/` directory in the path when hovering over a less filename in the inspector:
![Screenshot 2024-06-18 at 20 37 48](https://github.com/magento/magento2/assets/85479/246d4285-3712-4ed0-b6be-b2414ae4f21c)

### Questions or comments
No automated test were added, I don't think we have tests for grunt workflows, or do we?

### Contribution checklist (*)
